benchmarks/bench_plot_randomized_svd.py

```diff
@@ -323,8 +323,10 @@ def norm_diff(A, norm=2, msg=True, random_state=None):
 
 
 def scalable_frobenius_norm_discrepancy(X, U, s, V):
-    # if the input is not too big, just call scipy
-    if X.shape[0] * X.shape[1] < MAX_MEMORY:
+    # if the input is not sparse or sparse but not too big, call scipy
+    if not sp.sparse.issparse(X) or (
+        X.shape[0] * X.shape[1] * X.dtype.itemsize < MAX_MEMORY
+    ):
```

Comment:
Small nit:

```suggestion
    if not sp.sparse.issparse(X) or X.nbytes < MAX_MEMORY:
```

Reply:
I thought the same at one point, but `X.nbytes` does not work when `X` is a sparse matrix or `X` is a pandas DataFrame. Both happens in this benchmark.

Reply:
On closer inspection, the `not issparse(X)` changes the logic a little. With this PR:

1. If `X` is a dataframe -> Call SciPy
2. If `X` is a ndarray -> Call SciPy
3. If `X` is sparse and `X.shape[0]... < MAX_MEMORY` -> Call SciPy

I'm guessing, we want:

1. If `X` is a dataframe -> Call SciPy (The batching code does not work on dataframes)
2. If `X` is a ndarray and `X.size * X.itemsize < MAX_MEMORY` -> Call SciPy directly
3. If `X` is a sparse matrix and `X.size * X.itemsize < MAX_MEMORY` -> Call SciPy directly
4. Otherwise, batch.

Reply:
If `X` is a numpy array, then `X` fits in the RAM so `U.dot(np.diag(s).dot(V))` will fit in the RAM as well (there is a factor 2 but with `MAX_MEMORY=4e9` this should be OK), compute `A = X - U.dot(np.diag(s).dot(V))` and then call `scipy.linalg.norm(A)`.

The comment "Call Scipy" is slightly misleading, I think the potential blocker is creating the dense matrix `U.dot(np.diag(s).dot(V))` when matrices are sparse. If that is possible, calling `scipy.linalg.norm(A, norm="fro")` will not be an issue I think.

Reply:
I pushed a commit tweaking the comment and using `X.size` rather than `X.shape[0] * X.shape[1]`